src/zish4add.adoc

```diff
@@ -0,0 +1,36 @@
+== "Zish4add" Extension for 16-byte pointer addition, Version 0.9
+
+The B-extension adds `SH[1|2|3]ADD` instructions, and for RV64 all have
+an unsigned word .UW version for RV64.
+
+Zish4add adds two new analogous instructions to RV64: SH4ADD and SH4ADD.UW.
+
+[wavedrom, sh4add-wavedrom-reg,svg]
+....
+{reg:[
+    { bits:  7, name: 0x33, attr: ['OP'] },
+    { bits:  5, name: 'rd' },
+    { bits:  3, name: 0x7, attr: ['rv64: SH4ADD'] },
+    { bits:  5, name: 'rs1' },
+    { bits:  5, name: 'rs2' },
+    { bits:  7, name: 16, attr: ['rv64: SH4ADD'] },
+]}
+....
+
+Increment `rs2` by `rs1` shifted left by 4 bit positions and write the result to `rd`.
+
+
+[wavedrom, sh4adduw-wavedrom-reg,svg]
+....
+{reg:[
+    { bits:  7, name: 0x3b, attr: ['OP'] },
```

Comment:
OP->OP-32

Reply:
I think 0x3b is OP-32. sh4add says 0x33 and sh4add.uw says 0x3b, but they both label it as OP.

Reply:
ok - got it - sorry I thought that SH4ADD was wrong not SH4ADD.UW - I'll fix that now